include/fmt/base.h

```diff
@@ -1902,7 +1902,9 @@ template <typename Context> class basic_format_args {
     if (id >= detail::max_packed_args) return arg;
```

Comment:
I think the correct fix is to cast id to unsigned using a static cast to filter out negative out of bound values in addition to positive ones.